### PR TITLE
SHA1 is integrated into Stemcell

### PR DESCRIPTION
**Pull request by cunnie**
Many changes and enhancements to the continuous integration (anything under the `ci/` directory.

- Install awscli into docker image. (used to upload stemcell)
- Concourse has jobs for stemcell "families" (e.g. 3312.x, 3263.x, latest, etc...) <sup><a href="#note1">[Concourse Resource]</a></sup>
- Disabled manual triggering for publishing light stemcells.

<sup><a name="note1">[Concourse Resource]</a></sup>: We'll need switch back to using canonical resource after [PR is merged](https://github.com/concourse/bosh-io-stemcell-resource/pull/15) more finely-grained selection

Tracker story: [#134127235](https://www.pivotaltracker.com/story/show/134127235)

### CODE REVIEW

File: ci/stemcell/light/pipeline.yml [hunk 1]
```diff
@@ -2,72 +2,142 @@
 groups:
   - name: light-gce-stemcells
     jobs:
-      - ubuntu-trusty-stemcell
+      - ubuntu-trusty-stemcell-latest
+      - ubuntu-trusty-stemcell-3312
+      - ubuntu-trusty-stemcell-3263
+
+shared:
```
Comment (evandbrown) [inline ci/stemcell/light/pipeline.yml]: I haven't seen the `shared` key before. Is this something new(er) in Concourse? If so, maybe we can update the CI readme to reflect the minimum Concourse version.

File: ci/docker/bosh-google-cpi-boshrelease/Dockerfile [hunk 2]
```diff
@@ -50,6 +50,9 @@ RUN ruby-install ruby 2.1.2 && \
 # Install Bundler and BOSH CLI
 RUN /bin/bash -l -c "gem install bundler bosh_cli --no-ri --no-rdoc"
 
+# Install AWS CLI (used to upload stemcell)
+RUN pip install awscli
```
Comment (evandbrown) [inline ci/docker/bosh-google-cpi-boshrelease/Dockerfile]: Do we build and push the new Docker image after merge, or has that already been done?

File: src/bosh-google-cpi/integration/stemcell_test.go [hunk 3]
```diff
@@ -29,12 +53,24 @@ var _ = Describe("Stemcell", func() {
...
 			}`, stemcellCID)
+		assertSucceeds(request)
+	})
 
-		response, err := execCPI(request)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(response.Error).To(BeNil())
-		Expect(response.Result).To(BeNil())
-
+	//This currently fails due to issue: https://github.com/google/google-api-go-client/issues/181
```
Comment (evandbrown) [inline src/bosh-google-cpi/integration/stemcell_test.go]: Does the sha1 serve a purpose if it's not validated on disk creation? If not, then I'll find out more about when/if the validation will be implemented.

File: ci/stemcell/light/pipeline.yml (hunk 1 above)
Comment (cunnie) [inline ci/stemcell/light/pipeline.yml]: It's not a [Concourse](https://concourse.ci/pipelines.html#table-of-contents) key — the only top-level keys that Concourse has are `resources`, `resource_types`, `jobs`, and `groups`.

Rather, this is a generic YAML key that we use to place YAML anchors that we re-use throughout the pipeline. We could have just as easily called it `anchors:` or whatever.

If you prefer, we'd be happy to inline the anchors and remove the `shared` key.

File: ci/docker/bosh-google-cpi-boshrelease/Dockerfile (hunk 2 above)
Comment (cunnie) [inline ci/docker/bosh-google-cpi-boshrelease/Dockerfile]: Yes, we build and push the new Docker image after merge; we have not done that yet. Let us know when you've merged & we'll be happy to do it.

File: src/bosh-google-cpi/integration/stemcell_test.go (hunk 3 above)
Comment (cunnie) [inline src/bosh-google-cpi/integration/stemcell_test.go]: Correct, right now the SHA1 is unused as GCP does not support checksum validation yet. If you could find out the status of that feature, I'm sure that would help @cppforlife sleep a little easier :)

File: ci/docker/bosh-google-cpi-boshrelease/Dockerfile (hunk 2 above)
Comment (cunnie) [inline ci/docker/bosh-google-cpi-boshrelease/Dockerfile]: Oops — your comment exposed a flaw in our PR (we pointed to a stale Docker image/tag). We're gonna update the PR now. Thanks for the keen eyes!

**Comment by cunnie** [comment]
@evandbrown : We notice that many of the tasks are pointing to the v2 version of the boshcpi/gce-cpi-release docker image even though v3 was pushed 2 days ago.

Let us know if you want us to push v4 of the docker image (which will include our addition of the AWS CLI) and update the Concourse tasks to point to v4.

**Comment by zaksoup** [comment]
@evandbrown Hey, any movement on this?

**Comment by evandbrown** [comment]
@cunnie @zaksoup sorry for the delay on this. I'm merging now, and will build/push v4 and update the tasks when that's done. Thanks for the great work here!

**Comment by evandbrown** [comment]
Image is pushed and change to use v4 is in `develop`. Thanks again!

**Pull request merged**